client/src/components/geoJS/layers/measureToolLayer.ts

```diff
@@ -0,0 +1,259 @@
+import geo, { GeoEvent } from 'geojs';
+import { SpectroInfo } from '../geoJSUtils';
+import { LayerStyle, LineData, TextData } from './types';
+import BaseTextLayer from './baseTextLayer';
+
+function _determineRulerColor(isDragging: boolean, isDarkMode: boolean) {
+  if (isDarkMode) {
+    return isDragging ? 'orange' : 'yellow';
+  }
+  return isDragging ? 'cyan' : 'blue';
+}
+
+export default class MeasureToolLayer extends BaseTextLayer<TextData> {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  frequencyRulerLayer: any;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  pointAnnotation: any;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  lineAnnotation: any;
+
+  rulerOn: boolean;
+  dragging: boolean;
+  yValue: number;
+  hovering: boolean;
+
+  moveHandler: (e: GeoEvent) => void;
+  mousedownHandler: (e: GeoEvent) => void;
+  hoverHandler: (e: GeoEvent) => void;
+  mouseupHandler: () => void;
+
+  constructor(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    geoViewerRef: any,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    event: (name: string, data: any) => void,
+    spectroInfo: SpectroInfo,
+    measuring?: boolean,
+    yValue?: number
+  ) {
+    super(geoViewerRef, event, spectroInfo);
+
+    const textLayer = this.geoViewerRef.createLayer('feature', {
+      features: ['text']
+    });
+    this.textLayer = textLayer
+      .createFeature("text")
+      .text((data: TextData) => data.text)
+      .style(this.createTextStyle())
+      .position((data: TextData) => ({
+        x: data.x,
+        y: data.y,
+      }));
+
+    const frequencyRulerLayer = this.geoViewerRef.createLayer("feature", {
+      features: ['point', 'line'],
+    });
+    this.frequencyRulerLayer = frequencyRulerLayer;
+    this.rulerOn = false;
+    this.pointAnnotation= null;
+    this.lineAnnotation = null;
+    this.dragging = false;
+    this.yValue = yValue || 0;
+    this.color = 'white';
+    this.hovering = false;
+
+    this.textStyle = this.createTextStyle();
+    this.rulerOn = measuring || false;
+    if (this.rulerOn) {
+      this.enableDrawing();
+    }
```

Comment:
This (`enableDrawing`) needs to be moved down below the initialized of the handlerFunctions.  If the `ruleOn` is true when initializating the ruler you are initializing eventListeners on not defined values.  It may not error because eventually that memory location will have a function but I beleive geoJS will yell at you.  To replicate just watch the console when swapping between Compressed and Uncompressed views.